src/components/Quiz.js

```diff
@@ -18,19 +18,56 @@ const Quiz = () => {
     setQuestions(quizData);
   }, [quizName]);
 
-  const handleAnswerSelect = e => {
-    const selectedAnswer = { [e.target.name]: Number(e.target.value) };
+  const handleCheckboxAnswers = (e, answers) => {
+    const questionIds = Object.keys(answers || {});
+    const hasQPreviouslySelectedAnswers =
+      questionIds && questionIds.includes(e.target.name);
+
+    let checkedAnswer;
+    if (hasQPreviouslySelectedAnswers) {
+      const previousSelectedAnswers = answers[e.target.name];
+      const isAnswerPreviouslySelected = previousSelectedAnswers.includes(
+        Number(e.target.value),
+      );
+      if (isAnswerPreviouslySelected) {
+        const removePreviousAnswer = previousSelectedAnswers.filter(
+          answer => answer !== Number(e.target.value),
+        );
+        checkedAnswer = {
+          [e.target.name]: removePreviousAnswer,
+        };
+      } else {
+        checkedAnswer = {
+          [e.target.name]: [...previousSelectedAnswers, Number(e.target.value)],
+        };
+      }
+    } else {
+      checkedAnswer = { [e.target.name]: [Number(e.target.value)] };
+    }
+    return checkedAnswer;
+  };
+
+  const handleAnswerSelect = (e, type) => {
+    let selectedAnswer;
+    if (Object.keys(userAnswers).length && type === 'checkbox') {
```

Comment:
It looks like including `Object.keys(userAnswers).length` in the `if` statement is a bug. If the user is answering the very first question which is a checkbox, then `userAnswers` is still empty and we won't call the `handleCheckboxAnswers` function. Let's remove `Object.keys(userAnswers).length` from the `if` statement

Reply:
Done! :)